Add Practice Page for Supporting Color Contrast Settings

mcking65 opened the pull request:
Part of the resolution of issue #2864.

#### Preview Links

* [Preview revised practices page in compare branch](https://deploy-preview-480--aria-practices.netlify.app/aria/apg/practices/)
* [Preview color contrast settings practice page in compare branch](https://deploy-preview-480--aria-practices.netlify.app/aria/apg/practices/color-settings/)

___
[WAI Preview Link](https://deploy-preview-480--aria-practices.netlify.app/ARIA/apg) _(Last built on Thu, 06 Aug 2026 02:56:08 GMT)._

jongund commented:
@mcking65 
This is ready for review, I think it would be great if James Nurthen could review it.

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
change system color table

jongund commented:
@mcking65 
I added labels to the cells in the sample column of the System Colors table.

jongund committed:
updated links in testing

jongund committed:
fixed link

jongund committed:
updated intro

jongund committed:
updated intro

jongund committed:
updated intro

jongund committed:
updating testing

jongund committed:
updating testing

jongund committed:
updating testing

jongund committed:
added test page for contrast settings

jongund committed:
moved test content

jongund committed:
udpated content

jongund committed:
updated test page

jongund committed:
updated test page

jongund committed:
updated test page

jongund committed:
updated practices

jongund committed:
updated practice

jongund committed:
fixed spelling errors

jongund committed:
updated tables

jongund commented:
@mcking65 

I have been doing more investigation on supporting user high contrast settings.  There needs to be a little bit of a deep dive on this issue in the working group.   I added some feature tables to the current draft for high contrast support to help identify some of the issues we need to address in the practice and examples.  
The main issues are:

1. The author can on detect media queries values, so which media queries should our example support.  It is clear from the table `prefers-contrast: more` and `forced-colors:active` should be supported, but not sure about `prefers-color-theme`.  Note we don't have any examples that support `prefers-contrast: more` and only two that support `forced-colors: active`.

2. How mush we should discuss the "invert colors" option in operating systems, this is the first setting in most operating system accessibility options.  Preference more contrast is kind of buried in amongst other options.  Theme options are in appearance, not accessibility options.   Authors cannot detect it's use, so do we just ignore it in the practice?

I will be out of town on June 25th, but I should be there for the July 9th meeting.

jongund committed:
linting issue with link

jongund committed:
updated information on color contrsdt options

jongund committed:
updated information on user styling options

jongund committed:
updated descriptions

a11ydoer commented:
I will review this tomorrow. sorry for the delay.

a11ydoer commented:
Minor technical detail - In **System Colors** table, the computed sample column, "highlight," is not the same as the computed color, "#4d7198".  It seems to be #b3d7ff.

jongund committed:
fixed bug in computed color calculation

jongund committed:
updated system color table

jongund committed:
fixed bug

jongund committed:
fixed bug

jongund commented:
@a11ydoer 
Fixed bug in color calculation for system colors

a11ydoer commented:
> @a11ydoer Fixed bug in color calculation for system colors

<img width="884" alt="rgb color to hex color" src="https://github.com/user-attachments/assets/96b5d41b-d761-4801-9af1-bd0b6bfea810">
Hi @jongund RGBA for the "highlight color" is converted to hex value, #80bcfe99 when I used this website. I also attached the screenshot.

jongund commented:
@a11ydoer 

Jemma you need to add the alpha value (e.g. the 4 value in the RGB value) to compute the actual rendered color, the screen shot you sent only allows the R, G and B values to be used in the calculation.   The actual computed color depends on the color of the background, which in the case of the practices page is white (#ffffff).

mcking65 committed:
Merge remote-tracking branch 'origin/main' into high-contrast-practice

jongund committed:
Merge branch 'main' into high-contrast-practice

jongund committed:
updated

jongund committed:
Merge branch 'main' into high-contrast-practice

jongund committed:
editing practice

jongund committed:
updated content

jongund committed:
fixed html linting bug

jongund committed:
fixed html bugs

ariellalgilmore committed:
fix formatting

ariellalgilmore committed:
fix css linter issues

jongund committed:
testing push

jongund committed:
testing push

jongund committed:
updated css to remove diff info

mcking65 committed:
Editorial revisions to intro

mcking65 committed:
Link to WCAG min contrast requirement

mcking65 committed:
More editorial revisions to intro

mcking65 committed:
Editorial revisions to summary of operating system features.

mcking65 committed:
Satisfy HTML linter

mcking65 committed:
Consistency: change 'invert color setting' to 'invert colors setting' (most instances had the 's')

mcking65 committed:
Add tile to practices page

mcking65 committed:
Editorial revisions to invert colors section

jongund committed:
updated caption styling and documentation for increase contrast example

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund committed:
added links to table of high contrast options

jongund committed:
added new color scheme example

jongund committed:
updated headings for color scheme

jongund committed:
updated headings and example labels

jongund committed:
added links for Linux/GNOME contrast options

jongund committed:
Updated SVG vs. bit-mapped section

jongund committed:
some minor edting of content and getting caption color to match preview heading colors

jongund committed:
updated svg summary table

jongund committed:
updated section on chrome media query emulation features

jongund committed:
updated svg vs bit-mapped table

howard-e commented on .eslintrc.js:
Seems this has been added in error. We already have an eslint config at `eslint.config.mjs` and the changes here don't seem necessary to move across unless there's a specific reason to?

howard-e reviewed (commented):
Noticed some formatting requirements from the linter(s) that were been missed

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting errors

jongund committed:
fixed linting issue

jongund committed:
fixed linting issue

jongund committed:
fixed linting issue

howard-e commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+                  <li><a href="https://support.apple.com/guide/mac-help/change-display-settings-for-accessibility-unac089/mac">macOS</a></li>
+                  <li><a href="https://support.microsoft.com/en-us/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696">Windows</a></li>
+                  <li><a href="https://support.apple.com/en-ge/guide/iphone/ipheed8d2fca/18.0/ios/18.0">iOS</a></li>
+                  <li><a href="https%3A//support.google.com/accessibility/android/answer/11183305%3Fhl%3Den%23zippy%3D%252Cuse-high-contrast-text">Android</a></li>
```
```suggestion
                  <li><a href="https://support.google.com/accessibility/android/answer/11183305">Android</a></li>
```

@jongund the link is certainly valid and something to look more into about why this structure is invalid, but the suggestion gets to the same result, so it may be better to keep it trimmed.

Navigating to another link on that page and then navigating back yields additional session information in the URL for myself and I'm thinking that may be where `#zippy=` is coming from? (some special session value when this link was copied and pasted). Additionally, the `?hl=en` will be forcing the user to visit the page with it auto-translated to English which may not be what all users want.

jongund committed:
fixed liniting errors

jongund committed:
fixed liniting errors

jongund committed:
fixed liniting errors

jongund committed:
fixed liniting errors

jongund commented:
Can someone run the prettier code on the CSS to get rid of the CSS linting errors

howard-e committed:
Run npm run fix to address remaining lint errors

howard-e commented:
> Can someone run the prettier code on the CSS to get rid of the CSS linting errors

Done in https://github.com/w3c/aria-practices/pull/2991/commits/5720724d650679565184a2f0883c74557aa9456a

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+        <h3>Limitations of Bit-Mapped Images</h3>
+
+        <p>The colors of pixels used in bit-mapped images (e.g. .png, .jpeg) do not respond to changes in media query values making them difficult or impossible for people with some types of visual impairments to see when the change contrast settings.  Low resolution images also do not scale smoothly when the browser zoom features are used to increase the size of rendered content and the resultin ...
```
The first sentence isn’t entirely true anymore since authors can manipulate the appearance of bitmap images with CSS techniques like `filter: contrast()`, `mix-blend-mode`, etc. in reaction to a user preference media query.

jongund replied:
Ok, it should acknowledge this technique.   I will need to research the capabilities of the technique.

jongund replied:
@adampage 
I made some changes to the bit-mapped image section based on your feedback, could you review an comment?

adampage replied:
Thanks @jongund, these changes look great. 👍🏻

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+        <h4>CSS Media Query Code</h4>
+
+        <pre>
+          <code>
+@media (prefers-color-scheme: dark) {
+  button.color-scheme[role="switch"] {
+    background-color: #333;
+    color: #fff;
+  }
+
+  button.color-scheme[role="switch"] .label {
+    color: #fff;
+  }
+
+  button.color-scheme[role="switch"] svg rect {
+    fill: #36c;
+    stroke: #36c;
+  }
+
+  button.color-scheme[role="switch"] svg circle.off,
+  button.color-scheme[role="switch"] svg circle.on {
+    fill: #fff;
+    stroke: #fff;
+  }
+
+  button.color-scheme[role="switch"]:focus,
+  button.color-scheme[role="switch"]:hover {
+    border-color: #add8e6;
+  }
+}
+          </code>
```
Now that [`light-dark()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark) is widely implemented (only as of May 2024 😅), how about modernizing this code example using `color-scheme` and `light-dark()`? Or adding it as an alternative technique?

jongund replied:
Sounds like a good idea, want to look like we are keeping up with the times

jongund replied:
@adampage 
Maybe we could do this for the next publication cycle.

adampage replied:
Absolutely. 👍🏻

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+          User agents do not provide a media query that can determine whether invert colors is enabled in the operating system.
+        </p>
+        <p>
+          To support invert colors effectively, insure content meets
```
```suggestion
          To support invert colors effectively, ensure content meets
```

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+      <section id="prefers-contrast">
+        <h2>Increase Contrast</h2>
+        <p>When the increase contrast setting is is enabled the <code>prefers-contrast</code> media query changes from <code>no-preference</code> to <code>more</code>.
```
```suggestion
        <p>When the increase contrast setting is enabled the <code>prefers-contrast</code> media query changes from <code>no-preference</code> to <code>more</code>.
```

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+       <h3 id="prefers-contrast-example">Increase Contrast Example</h3>
+
+        <p>The example shows how the color contrast was increased from 12.8 to 18.1 for text and 4.6 to 12.2 for border and fill for the button when the <code>prefers-contrast: more</code> is set.  The example simulates the change on colors when the increased option switch is set to "on".  If you enable operating system "increased contrast" feature the example will use be rendered in any state wi ...
```
```suggestion
        <p>The example shows how the color contrast increases from 12.8 to 18.1 for text and 4.6 to 12.2 for border and fill for the button when the <code>prefers-contrast: more</code> is set.  The example simulates the change to colors when the Increase Contrast option switch is set to "on".  If you enable operating system "increased contrast" feature the example will be rendered with the increased contrast values.</p>
```

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+        <div class="example">
+
+            <button type="button" role="switch" aria-checked="false" class="increase-contrast">
+              <span class="label">Increase Contrast</span>
+              <svg xmlns="http://www.w3.org/2000/svg" height="20" width="36">
+                <rect class="container" x="1" y="1" width="34" height="18" rx="9" />
+                <circle class="off" cx="11"  cy="10" r="8" />
+                <circle class="on"  cx="25" cy="10" r="8" />
+              </svg>
+              <span class="on" aria-hidden="true">On</span>
+              <span class="off" aria-hidden="true">Off</span>
+            </button>
```
I’m not seeing any change to the page when I toggle either the Increase Contrast or Dark Color Scheme switches. Are those meant to immediately show a simulated effect?

jongund replied:
Yes, there are suppose to change.
It may be there should be a separate switch to tun on the emulation next to the example.

adampage replied:
Thanks @jongund, this is clearer. However, because the examples themselves _**are**_ toggle switches, I wonder if it might still invite a bit of confusion for some readers since the concepts of dark mode and high contrast are so often expressed as interactive toggles. 😅 

If the examples were just some static rich text, like a heading, a paragraph, and some sort of colored embellishment (like a blockquote or bulleted list), then I think they would be crystal clear — especially for the users who accept our invitation to enable their own device settings to observe the reaction.

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+       <h3 id="color-scheme-switch">Color Scheme Example: Switch</h3>
+
+        <p>The example shows how colors can change when users switch color schemes.  The example simulates the change of colors when the dark color theme is enabled when the switch is set to "on".  If you enable operating system "dark scheme" feature the example will be rendered in any state with the dark color scheme.</p>
```
```suggestion
        <p>The example shows how colors can change when users switch color schemes.  The example simulates the change of colors when the dark color theme is enabled when the switch is set to "on".  If you enable the operating system’s "dark scheme" feature the example will be rendered with the dark color scheme.</p>
```

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+        <h3 id="color-scheme-wikipedia">Color Scheme Example: Wikipedia Page</h3>
+
+        <p>The following example illustrates how Wikipedia supports color scheme media query.  The example includes showing the "Appearance" sidebar allowing the user to choose the light or dark scheme and other rendering options for text size and column width.</p>
```
```suggestion
        <p>The following example illustrates how Wikipedia supports the color scheme media query.  The example includes showing the "Appearance" sidebar allowing the user to choose the light or dark scheme and other rendering options for text size and column width.</p>
```

adampage commented on content/practices/high-contrast/high-contrast-practice.html:
```diff
@@ -0,0 +1,779 @@
+
+        <h3>Testing Dark Color Scheme</h3>
+
+        <p>Turn on the dark color them feature for major operating systems and verify text and components color contrast ratios meet at least <a href="https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum">WCAG 1.4.2: Contrast (Minimum)</a> requirement.</p>
```
```suggestion
        <p>Turn on the dark color theme feature in major operating systems and verify text and components satisfy the <a href="https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum">WCAG 1.4.2: Contrast (Minimum)</a> success criterion.</p>
```

Very editorial. 😅

adampage reviewed (changes requested):
This reference will be very helpful to many people! 🤩

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/high-contrast/high-contrast-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
updated section on bit-mapped image to include the possibility of filters or image substitution

jongund committed:
updated switch examples

jongund committed:
fixed liniting bug

jongund committed:
fixed liniting bug

jongund committed:
updated switch example labels

jongund committed:
emphiszed actual colors in system color section

jongund committed:
updated summary table for bit-mapped vs SVG images

mcking65 committed:
editorial changes in increase contrast section

mcking65 committed:
editorial changes to increase contrast testing section

mcking65 committed:
improve description of increase contrast

mcking65 committed:
editorial revisions to dark mode testing

mcking65 committed:
editorial revisions to dark mode example section

mcking65 committed:
More editorial revisions to dark mode example section

jongund committed:
updated switch example to use light-dark function

jongund committed:
updated switch example to use light-dark function

jongund committed:
editorial changes to color scheme description

jongund committed:
edited high contrast description

jongund commented:
@mcking65 @howard-e 
The preview is still out of date with what's in the prepository

howard-e commented:
> @mcking65 @howard-e The preview is still out of date with what's in the prepository

@jongund the referenced link in the [top comment](https://github.com/w3c/aria-practices/pull/2991#issue-2242723541) had to be completely rebuilt so it's a new [link](https://deploy-preview-380--aria-practices.netlify.app/ARIA/apg) now. I can only assume the older link was checked. My apologies for not clarifying that in the previous meeting.

The changes of your [most recent commit](https://github.com/w3c/aria-practices/pull/2991/commits/61c08444faa6ace5e933b68d1c6b20ed014bfc0d) under the [Color Scheme (Light or Dark) `h2`](https://deploy-preview-380--aria-practices.netlify.app/aria/apg/practices/high-contrast/) are present to me when viewing that updated link. Please let me know if this matches what's expected.

jongund committed:
added system color warning

jongund committed:
added cautions

jongund committed:
updated caution styling

jongund committed:
added caution for system colors and currentcolor keyword

jongund commented:
@mcking65 @howard-e 
The preview link is failing with some updates from today.

jongund commented:
@mcking65 @howard-e 
The preview is being built now, just different preview number

jongund committed:
updated caution for system colors

jongund committed:
removed confusing content

jongund committed:
edited table caption

jongund committed:
updated example for current color

jongund committed:
fixed spelling error

jongund committed:
update current color example

jongund commented:
@howard-e 
Any ideas on why the build is failing?

jongund committed:
updating guidance for current color

jongund commented:
@mcking65 
I am working on some changes, so please wait to review until I commit my changes problably on Monday.

jongund committed:
building table of color changes

jongund committed:
added additional elements

jongund committed:
added additional elements

jongund committed:
updated content

jongund committed:
contrast theme images

jongund committed:
update information on contrast themes

jongund committed:
update

jongund committed:
updated switch example using system colors

mcking65 committed:
rename from high contrast to color settings

mcking65 commented:
@jongund 

Heads up that when I changed the title as agreed in the last meeting, I renamed the directory and main file. I am making some of the other editorial changes we agreed to make.

mcking65 committed:
update descriptions of settings based on last week's discussions

mcking65 committed:
Update H2 text

mcking65 committed:
fix spelling

mcking65 commented:
@jongund  could you please fix the linting errors. Apparently there is a bad link someplace too.

@howard-e the preview build is failing again. Wonder if it has to do with the directory renaming I did.

jongund committed:
update file naming to use color-settings

jongund committed:
updated example headings

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

howard-e commented on content/practices/practices.html:
```diff
@@ -75,6 +75,18 @@ <h2 class="tile-name">
       </li>
 
+      <li class="tile">
+        <a href="high-contrast/high-contrast-practice.html">
```
```suggestion
        <a href="color-settings/color-settings-practice.html">
```

@mcking65 needed change because of the folder rename in https://github.com/w3c/aria-practices/pull/2991/commits/3d7cc355895bb34cb6b391143cf2f516017c60f1

jongund committed:
updated system color switch example

howard-e commented:
> @howard-e the preview build is failing again. Wonder if it has to do with the directory renaming I did.

This should be resolved now. And the updated content is available at https://deploy-preview-380--aria-practices.netlify.app/aria/apg/practices/color-settings/

howard-e commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1055 @@
+      <section id="forced-colors">
+        <h2>Contrast Themes (Forced Colors)</h2>
+
+        <p>The <code>forced-colors</code> CSS media query provides a means for components to use the color preferences of people with visual impairments.   When the user chooses a contrast theme in the operating system (i.e. in Windows 11 Accessibility Settings > Contrast Theme), browsers set the <code>forced-colors</code> property to <code>active</code>.  When forced colors are active the browse ...
```
```suggestion
        <p>The <code>forced-colors</code> CSS media query provides a means for components to use the color preferences of people with visual impairments.   When the user chooses a contrast theme in the operating system (i.e. in Windows 11 Accessibility Settings &gt; Contrast Theme), browsers set the <code>forced-colors</code> property to <code>active</code>.  When forced colors are active the browser overrides any author styling and applies the system color associated with the theme to each element.   Authors need to match the styling preferences of users with the semantics features of their components. For example text used in custom components for labels and instructions should use the same colors as the users contrast theme for text content.  Components that behave as links need to match the contrast theme styling for links and other interactive custom components should be based on the styling used for buttons, textbox and range controls for their features.</p>
```

The above suggestion will fix the HTML linting. And running `npm run fix` should address the remaining CSS listing issues

mcking65 committed:
Fix link and description for color settings practice

mcking65 committed:
Merge branch 'high-contrast-practice' of https://github.com/w3c/aria-practices into high-contrast-practice

mcking65 committed:
editorial refinement

mcking65 committed:
do not imply "authors need" to provide support.

mcking65 committed:
HTML lint fix from Howard

Co-authored-by: Howard Edwards <howarde.edwards@gmail.com>

mcking65 commented:
Thank you @howard-e!

I updated the preview links section in the top comment.

howard-e committed:
Fix css lint errors

jongund committed:
Updated switch current color

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

mcking65 committed:
editorial revision to practices page card

mcking65 committed:
revise intro

mcking65 committed:
revise descriptions

jongund committed:
updated information on currentcCOlor and system colors

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund committed:
updated system color table to include checkboxes

mcking65 committed:
Editorial consistency of contrast theme practice cell in table

mcking65 committed:
Merge branch 'high-contrast-practice' of https://github.com/w3c/aria-practices into high-contrast-practice

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1197 @@
+              </td>
+              <td>
+                When a user selects a contrast theme, the CSS media query <code>forced-colors</code> is set to <code>active</code>.
+                Use <code>currentcolor</code> and <code>system-color</code> values to style user interface controls and custom widgets to the user preferences.
```
@jongund  given what you have been saying about currentcolor, I am assuming you want to change this line? Maybe just delete the part about currentcolor?

mcking65 commented:
@jongund 

I revised intro and user options table based on today's discussion. Please review for accuracy.

jongund committed:
updated content

jongund committed:
updated guidance for forced-colors

jongund committed:
updated screen shot images

jongund committed:
added defalt images

jongund committed:
update switch using system colors

jongund committed:
added content related to div based switch example

jongund committed:
updated documentation

jongund committed:
updated system color guidance and screenshots

jongund committed:
updated system color table

jongund committed:
updated comparison tables

jongund committed:
updated comparison tables

jongund committed:
updated table

jongund committed:
added color theme data

jongund committed:
updated switch comparison

mcking65 committed:
Fix spelling errors

mcking65 committed:
minor editorial change to intro

mcking65 committed:
editorial consistency in invert colors section

mcking65 committed:
Editorial revisions to increased contrast section.

mcking65 committed:
fix spelling of accent

mcking65 commented:
@jongund 
There are quite a few lint errors. Can you clean them up?

@howard-e 
The preview is not showing latest commits again. Is that because of the failing checks?

jongund committed:
updated switch example comparison

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund committed:
editing labels

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

jongund committed:
fixing linting errors

howard-e commented:
> @howard-e The preview is not showing latest commits again. Is that because of the failing checks?

@mcking65 The expected changes are there now without my needing to do anything. It may have been a one-off error at the time last checked. These present lint errors don't impact the preview build

jongund committed:
fixed liniting errors

jongund committed:
fixed html liniting error

jongund committed:
editorial changes to light dark media queries

jongund committed:
made some editorial changes to content

jongund committed:
fixed spelling error

jongund committed:
removed unused images

adampage reviewed (changes requested):
Heya @jongund, just finished reviewing the reworked Contrast Themes section. Lots of great new info. 👍🏻  I added a few new suggestions.

jongund committed:
Update content/practices/color-settings/color-settings-practice.html

Looks good to me.  Thank you for reviewing and suggestions.

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/color-settings/color-settings-practice.html

Looks good to me.  Thank you for reviewing and suggestions.

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/color-settings/color-settings-practice.html

Looks good to me.  Thank you for reviewing and suggestions.

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/color-settings/color-settings-practice.html

Looks good to me.  Thank you for reviewing and suggestions.

Co-authored-by: Adam Page <adam@adampage.net>

jongund committed:
Update content/practices/color-settings/color-settings-practice.html

LGTM

Co-authored-by: Adam Page <adam@adampage.net>

jongund commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1546 @@
+
+        <pre>
+          <code>
+@media (forced-colors: active) {
+  div[role="switch"].system-color {
+    background-color: ButtonFace;
+    color: ButtonText;
+  }
+
+  div[role="switch"].system-color .label {
+    color: ButtonText;
+  }
+
+  div[role="switch"].system-color svg rect {
+    stroke: ButtonBorder;
+  }
+
+  div[role="switch"].system-color svg circle.off {
+    stroke: ButtonBorder;
+    fill: GrayText;
+  }
+
+  div[role="switch"].system-color[aria-checked="true"] svg circle.on {
+    stroke-color: ButtonBorder;
+    fill: ActiveText;
+  }
+
+  div[role="switch"].system-color:focus,
+  div[role="switch"].system-color:hover {
+    background-color: ButtonFace;
+  }
+}
```
I'll leave this editorial decision to Matt King.  
It would be the first use of this format in ARIA examples.  
I have no reason to object to the change.

smhigley replied:
I've been writing CSS almost exclusively with pre- or postprocessors for something like a decade now, so the nesting looks more friendly and easier to read to me personally :D

That said, I feel like I maybe no longer have a handle on what the community at large and especially newer devs would find easier to read.

jongund commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1546 @@
+
+        <h3 id="computed-system-colors">System Color Support</h3>
+
+        <p>The following table identifies the system colors defined in <a href="https://www.w3.org/TR/css-color-4/#css-system-colors">CSS Color Module Level 4</a> that support contrast themes.  When a user enables a contrast theme not all system colors change to use one of the colors defined in the theme.   The following table identifies which system colors support contrast theme colors.  Each sy ...
```
I think we need to some more edits to line 1305.  It is important to point out that the example colors are computed in real time based on the current browser and operating system settings.   So if the person viewing the table changes a system setting  that effects system colors they will see the effect in the table immediately.  I hope this makes sense.

jongund commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1546 @@
+
+        <p>Current color can be used when the custom component is based on existing HTML elements and the system color matches the semantics of the features of the component.  For example, the following section includes an switch component example using the <code>button</code> element.   The <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/color_value"><code>currentcolor</code> value</a> ...
+
+        <p>System colors values should be used when the custom elements are built using elements whose current color does not match the semantics of the custom component and when <code>currentcolor</code> is insufficient for matching the styling needs of the component to make it's features discernible.  Using System colors allows for setting both foreground and background colors and making featur ...
```
Sure that sounds good to just use `currentColor`

mcking65 committed:
Update content/practices/color-settings/color-settings-practice.html

Co-authored-by: Adam Page <adam@adampage.net>

mcking65 committed:
Format to support line commenting

mcking65 committed:
attempt some clarification

mcking65 commented:
@howard-e 

The preview timestamp updates, but the preview page is still not reliably updating.

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1568 @@
+        </p>
+
+        <p>
+          Authors can use either <code>currentcolor</code> or <code>&lt;system-colors&gt;</code> CSS values for components to adapt to appropriate colors in the contrast theme.
```
This is the first mention of currentcolor. It seems out of place. At this point, we have some explanation of the fact that the browser uses system colors for replacement colors, but no understanding of how currentcolor is relevant to authors. We don't have any explanation of currentcolor, and we don't seem to have a link to MDN for it.

howard-e commented:
> @howard-e
> 
> The preview timestamp updates, but the preview page is still not reliably updating.

There was bad netlify cache in the generated build that needed a refresh. It's resolved now

adampage reviewed (changes requested):
Hi @mcking65, here are suggestions for the remaining CSS blocks as [we chatted about in today’s APG meeting](https://www.w3.org/2025/02/18-aria-apg-minutes#49aa).

mcking65 committed:
Use nested CSS

Co-authored-by: Adam Page <adam@adampage.net>

mcking65 committed:
Use nested CSS

Co-authored-by: Adam Page <adam@adampage.net>

mcking65 committed:
Adam's revision that implements native CSS nesting

Co-authored-by: Adam Page <adam@adampage.net>

mcking65 committed:
Adam's revision that implements native CSS nesting

Co-authored-by: Adam Page <adam@adampage.net>

mcking65 commented:
@jongund 

After reviewing the resources that @smhigley mentioned in our last meeting, I think we need some more revisions before publishing. There are some important ways in which the current draft is incomplete or potentially incorrect.

For example, the section on invert colors incorrectly states that authors cannot detect the setting. However, one of the pages referenced in Sarah's article mentions [inverted-colors - CSS: Cascading Style Sheets | MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/inverted-colors).

Useful references:
1. [Assistive technology: Operating System and Browser Accessibility Display Modes - The A11Y Project](https://www.a11yproject.com/posts/operating-system-and-browser-accessibility-display-modes/)
2. [Styling for Windows high contrast with new standards for forced colors - Microsoft Edge Blog](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/)
3. [Quick Tips for High Contrast Mode | Sarah Higley](https://sarahmhigley.com/writing/whcm-quick-tips/)
4. [WHCM and System Colors — Adrian Roselli](https://adrianroselli.com/2021/02/whcm-and-system-colors.html#WCAGUpdate)

jongund committed:
updated practices based on feedback, including removing discussion of current color, since I think system colors should be used instead and easier to understand

jongund committed:
added contrast them test page

jongund committed:
updated test page results

jongund committed:
updated test documentation

jongund committed:
updated current color

jongund committed:
fixed bugs in test page

jongund committed:
fixed bugs in test page

jongund committed:
udpate practices

jongund committed:
udpate practices

jongund committed:
udpate practices

jongund committed:
udpate practices

howard-e commented:
> The preview is not building, what do I need to do to get it working again?

@jongund the error happened because the `test-contract-theme.html` file was unexpected. The error is as follows: 

<details>

<summary>expand error</summary>

```sh
Error: Could not determine content type for file at /home/runner/work/wai-aria-practices/wai-aria-practices/_external/aria-practices/content/practices/color-settings/test-contrast-theme.html
This file may not follow established conventions.
    at determineContentType (/home/runner/work/wai-aria-practices/wai-aria-practices/scripts/pre-build/library/determineContentType.js:55:9)
```

</details>

The preview has now built properly. I've created a temporary solution through https://github.com/w3c/wai-aria-practices/pull/388 to make handling this easier and push this PR forward.

But it does highlight the brittleness of this approach so we'll eventually need to move such a configuration to this repository as the ideal solution, which I'm tracking in https://github.com/w3c/wai-aria-practices/pull/389. 

cc @mcking65

jongund commented:
@howard-e 
Don't want to make a lot of extra work, so I'll just remove the test file from the repository and put it somewhere else on the web.

jongund committed:
removed test files

jongund commented:
@howard-e 
I have removed the test files from the repository

jongund committed:
update current color for SVG

jongund committed:
updated SVG current color content

jongund committed:
updated contrast theme section

jongund committed:
fixed broken links

jongund commented:
@mcking65 
I have made some updates to invert colors and the contrast theme section.

jongund commented:
@howard-e 
Hi howard the preview didn't seem to build, what do I need to do?

howard-e commented:
> @howard-e Hi howard the preview didn't seem to build, what do I need to do?

@jongund nothing required of you here. But an infrastructure issue that needs to be investigated. The [updated changes](https://deploy-preview-380--aria-practices.netlify.app/aria/apg/practices/color-settings/) are available

jongund committed:
update practices

jongund committed:
add system colo note

jongund committed:
updated note in browser support table

jongund committed:
updated computed color sentences

jongund committed:
added link to note on unsupoorted system colors

jongund committed:
added information about computed color changing when contrast theme changes

jongund committed:
slight edit to sentenace

jongund committed:
added only Windows 10 and 11 support contrast themes

jongund committed:
Updated with changes to main
Merge branch 'main' into high-contrast-practice

jongund committed:
Merge branch 'main' into high-contrast-practice

jongund committed:
udpate color settings practice

jongund commented:
@mcking65 
I have edited the color settings to simplify the section on supporting Windows high contrast settings.

jongund committed:
Merge branch 'main' into high-contrast-practice

jongund committed:
fixed gammar issue

mcking65 committed:
Merge remote-tracking branch 'origin/main' into high-contrast-practice

jugglinmike commented:
The ARIA Authoring Practices (APG) Task Force just discussed this issue.

<details>
  <summary>The full log of that discussion</summary>

Matt_King: This is a giant pull request, and it’s been open for a long time. I recently merged the “main” branch back into it to make sure there are no conflict
Matt_King: It may be trying to do too much
Matt_King: jongund and I are the primary authors. jongund did a lot to pull together the relevant information, including a survey of blog posts. We were trying to hit the bar of being practical.
Matt_King: I think what might be most helpful right now is to get some general feedback right now. High-level feedback on whether it is organized in a way that we believe will be helpful to a reader who is interested in this topic. Also, are we covering useful information?
Matt_King: I’m trying to figure out: what’s the best way to move this forward? I don’t want to put something in APG if it’s hard to maintain but not actually useful.
jongund: My biggest concern is that in the APG, we don’t do much of any of this.
jongund: Putting it in makes it look like it’s important to do, but we don’t do it
jongund: We have some examples that demonstrate doing some of these things
jongund: But from a practicality standpoint: should I implement contrast?
Matt_King: That’s a good point. What we would do in the APG or what we wouldn’t do in the APG does suggest something about the importance of each item
Matt_King: Are these things that people don’t do because they don’t have to? Or because they don’t prioritize it?
Matt_King: What should the APG cover in this space?
jongund: People see these things, so we should say something about them. Saying nothing discounts them to other people. Readers will see these things out there and wonder if they should do something about them.
Adam_Page: At Hilton, we are generally pretty aware and savvy about the techniques that are described here. We use them as little as possible because we want things to just work”. We don’t want them to get in the way of default behavior
Adam_Page: We do have some pretty specific scenarios where we use “high contrast” mode, eg. in our “tab panel” implementation (our “active tab” includes some high-contrast-mode specificity to ensure that the visual indication of the active tab shows up there)
Matt_King: is that in the system colors?
Adam_Page: Yes
Matt_King: Another company out there that doesn’t have your awareness is including things based on the APG. If you’re doing styling that disappears if someone is using these accessibility settings, then this is important. That’s kind of a prioritization statement.
Matt_King: Are there any automated checks out there that say, “you ARE doing automated styling” and that will disappear if these features are in use?
Adam_Page: Not that I’ve seen.
Matt_King: Computed style is available to all the checkers
Adam_Page: Yes, but I haven’t seen any checker that attempts to deduce that the style is informative
Matt_King: Ah, right. Recognizing the problems from specific properties applied in specific ways is not necessarily deterministic.
Adam_Page: right
Adam_Page: I think this is really useful. It’s interesting that we’re considering it in APG because ARIA figures into it so lightly
Matt_King: Yes, that’s true.
Adam_Page: It’s probably no coincidence that the one place at Hilton where we DO do something is in an ARIA feature (the tab-set). Otherwise, we try to rely on native semantics.
Matt_King: I wonder if it would be useful to state something along those lines in the into
Matt_King: How do you do something at the top of this that say, “for you, this is interesting and here’s why” so that for everyone else, they recognize that it’s safe to ignore the page.
Matt_King: So from your point of view, Adam_Page, this still seems potentially valuable in the APG
Adam_Page: Definitely
jongund: What about color scheme (light/dark modes)?
Adam_Page: Hilton doesn’t support them in our web applications (though we do in our native mobile apps)
jongund: Does the W3 support it?
daniel: The specifications do. I’m not sure about the WAI website
joe: I think it would be useful to see these considerations in one place. I’ve seen each topic in separate areas, so collecting them together is great.
joe: Having considerations against each method (perhaps in a new column of the table) would be nice
joe: It would be nice to have “things to look out for” when using high-contrast mode. Just to sort of walk people through that process.
Matt_King: Can that table handle another column? It feels wide to me, but I can’t see it.
joe: Visually, it does look quite large on the page. I’m not sure
Matt_King: Were you suggesting that near the beginning, we add some text?
joe: Something about what people should check for when it comes to what visual information will be perceivable
Adam_Page: On the “quantity spin button” example I built last year, I did do some high-contrast work. I don’t remember specifically what I did, though
Adam_Page: Quantity Spin Button example and mention of forced colors https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/examples/quantity-spinbutton/#:~:text=When%20forced%20colors%20are%20enabled%2C%20system%20color%20keywords%20are%20used%20to%20ensure%20visibility%20and%20sufficient%20contrast%20for%20the%20spin%20button%E2%80%99s%20content%20and%20interactive%20states.
Matt_King: What would the problem be if people did NOT do that? Is that the kind of guidance we’re looking for?
Adam_Page: I think one active state might not show up
Matt_King: The fact that you can’t remember even after reading what we’ve written is a little concerning. Maybe we should update the wording to describe the specific situation
jongund: There are some broken parts to contrast themes, e.g. with pseudo-elements
Matt_King: It sounds like there is some value in continuing this work
Matt_King: It sounds like we need some kind of introductory statement that describes when/how this is useful
Matt_King: And a comment on the “quantity spin button” that explains “this is what would have happened if we didn’t implement high-contrast” would be helpful
jongund: There is a section at the end about SVG versus bitmap graphics
Matt_King: Maybe that doesn’t belong. That may be another future discussion
Adam_Page: I have time to read through this.
Adam_Page: I can start by doing exactly what you suggested regarding adding some explanatory text (it has to do with visually making three buttons appear as one. I used “Highlight” to do that (https://github.com/w3c/aria-practices/blob/main/content/patterns/spinbutton/examples/css/quantity-spinbutton.css#L17). It may have been somewhat uninformed at the time)
Matt_King: If we have an example like this, in other parts of the article, we have screenshots to depict what the failure state looks like in high-contrast mode (along with screenshots to depict the “normal” visual rendering). That demonstrates the kind of problem that authors need to be aware of
Matt_King: If we could knock this one out in the next couple months, it would be great to get it out of the backlog!

</details>

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+              <td>
+                <ul>
+                  <li><a href="https://support.apple.com/guide/mac-help/change-display-settings-for-accessibility-unac089/mac">macOS</a></li>
+                  <li><a href="https://support.microsoft.com/en-us/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696">Windows</a></li>
```
As far as I can understand, Windows only supports forced colours, and there is no increased contrast mode - therefore, I think we should either remove this list item, or explicitly state that it is not currently supported on Windows.

jongund replied:
I'll remove

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+                <ul>
+                  <li><a href="https://support.apple.com/guide/mac-help/change-display-settings-for-accessibility-unac089/mac">macOS</a></li>
+                  <li><a href="https://support.microsoft.com/en-us/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696">Windows</a></li>
+                  <li><a href="https://support.apple.com/en-ge/guide/iphone/ipheed8d2fca/18.0/ios/18.0">iOS</a></li>
```
This link seems to redirect to the documentation for changing the contrast specifically for the “Magnifier” app relating to the content displayed by the camera feed. 

I wonder if the: [iPhone: Make iPhone text easier to read with accessibility features](https://support.apple.com/guide/iphone/make-text-easier-to-read-iph3c076905a/17.0/ios/17.0) from the end of the article, could be used here instead?

jongund replied:
Updated link text an referene, there does not appear to be a specific link on the apple website identifying this feature.

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+                  <li><a href="https://support.apple.com/guide/mac-help/change-display-settings-for-accessibility-unac089/mac">macOS</a></li>
+                  <li><a href="https://support.microsoft.com/en-us/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696">Windows</a></li>
+                  <li><a href="https://support.apple.com/en-ge/guide/iphone/ipheed8d2fca/18.0/ios/18.0">iOS</a></li>
+                  <li><a href="https://www.android.com/accessibility/vision/">Android</a></li>
```
Similar to the Windows item, I don't think there's a way to specifically increase contrast in a way that would trigger the `prefers-contrast` query on Android, so I wondered if we might want to remove this list item, or state that it’s not supported?

jongund replied:
I'll remove

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+            Instead of shadow styling use highlights and gradients, since shadow styling is difficult to perceive in dark mode.
+          </li>
+          <li>
+            Use CSS filters on images to reduce their brightness to make them easier to view in dark mode (*** is this advice we want to give? ***).
```
I’m unsure about this. I understand that the reasoning for this is to help people who have a form of light sensitivity. However, I don’t think that there’s a way to do so, in an approach that is consistently usable, without risking making images difficult to perceive. Therefore, I think that this article should not feature this suggestion. Instead, I think the best approach may be for people to manually adjust their screen brightness through their OS settings.

jongund replied:
I think we just get rid of this paragraph.  It is really identifying an anti-pattern, which we avoid doing in the practices.

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+          browsers set the <code>forced-colors</code> property to <code>active</code>.
+          When using ARIA to alter the semantics of elements authors need to verify the appropriate system color for each feature when forced colors are enabled.
+
+          The following table provides guidance on how system colors should be applied when forced colors is enabled for common user interface controls.  When elements with altered semantics do not match the appropriate system color for elements, authors must set the appropriate system color using the <code>forced-colors: active</code> media query.
```
I checked this a few times, and I wonder if this line is relating to the "Browser System Color Support" table later on in the page. Maybe it's been moved since this content was first written, as the next table after this text relates to the switch example. Therefore, I wondered if we wanted to add a link in this text, to allow people to jump to the "Browser System Color Support" table.

jongund replied:
If you were looking in the pull request you won't see it, it is a generated table that is visible in preview mode.

joelamyman commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,946 @@
+            Most interactive custom components should be using the <code>buttonBorder</code>, <code>ButtonFace</code> and <code>ButtonText</code> values.
+          </li>
+          <li>
+            For more complex interactive components to differentiate component features consider using <code>Highlight</code> or <code>ActiveText</code> values to make the features easier to identify. See the <a href="#html-input-range-contrast-theme">HTML range slider</a> (e.g. <code>input[type="range"])</code>) for how other colors than just button colors are used in styling the slider and thum ...
```
The “HTML range slider” link doesn’t seem to redirect anywhere. I wasn’t sure where this should point to, or whether this content needs removing?

jongund replied:
I have removed this section, I do not think we should do to much about using media queries for custom colors, do many bugs.

joelamyman reviewed (commented):
I *think* I put myself forward to have a read through this and provide any feedback, although I can't see that in the minutes, so please feel free to dismiss this feedback if I've got this issue confused with another 😅 

This is a really great piece and does such a great job at consolidating these different colour-related settings, and providing guidance around how they work, how to develop for them, and how to test them. I think this is a fantastic resource, and would be great for people trying to understand the topic.

I've left a few minor points of feedback, but one other thing I wanted to mention is around the structure of the document. It seems like it might have changed a few times with content moving around a little bit, and is ever so slightly inconsistent.

When we looked at the page during the previous APG task force call, I thought it could do with information about authoring practices for each colour setting, however, after reviewing the page I now see that this is not needed as every section has this.

To help make the structure more clear, I wonder if each section could be separated in to:

```
<h2><Name of the setting></h2>
  <h3><Name of the setting> Example</h3>
  <h3>Supporting <Name of the setting></h3>
  <h3>Testing <Name of the setting></h3>
```

I note that most sections follow this, but having it consistent would make this really clear. 

For the suggested “Supporting…” heading, I don't think this needs additional content, but my suggestion is more to move any applicable content from each setting’s intro paragraph, and put it under a dedicated heading to make it easy to find, given how important it is.

To this effect, I also wondered if the “SVG graphics” section and “Testing for Contrast Theme Support” headings, could fall under the “Contrast Themes” `h2` to also help improve the structure.

mcking65 commented:
@joelamyman  really like your suggestion for the structure of the content.

jongund committed:
started reorginaizing color setting practice

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund committed:
updated practices page base don feedback from Joel

jongund committed:
fixed linting errors

jongund committed:
update SVG heading

jongund committed:
moved references tto their own section

daniel-montalvo commented:
Discussed in [29 April meeting](https://www.w3.org/2026/04/29-ARIA-APG-minutes.html#5aa8)

jongund committed:
Merge branch 'main' into high-contrast-practice

joelamyman reviewed (approved):
Apologies for the delay in providing review feedback. I've reviewed the latest version of this page, and it looks really great! I think the article flows really well, and to have all of these different contrast considerations in one place, makes this a fantastic resource! Thanks for implementing my feedback where appropriate @jongund! There's nothing else I can think of, this looks great to me!!

jongund committed:
Merge branch 'main' into high-contrast-practice

jongund committed:
took out css filters

mcking65 committed:
Merge remote-tracking branch 'origin/main' into high-contrast-practice

mcking65 committed:
fix lint error

mcking65 commented:
@jongund 

I was reviewing the changed files list and noticed unused images. Please remove them. I had an agent make a list of unused files, and it came up with the following. I only spot checked the agent's work, so this list may contain errors.

Unreferenced added images:

1. content/practices/color-settings/images/a-element-aquatic-theme.png
1. content/practices/color-settings/images/a-element-desert-theme.png
1. content/practices/color-settings/images/a-element-night-sky-theme.png
1. content/practices/color-settings/images/button-element-aquatic-theme.png
1. content/practices/color-settings/images/button-element-desert-theme.png
1. content/practices/color-settings/images/button-element-night-sky-theme.png
1. content/practices/color-settings/images/checkbox-element-aquatic-theme.png
1. content/practices/color-settings/images/checkbox-element-desert-theme.png
1. content/practices/color-settings/images/checkbox-element-night-sky-theme.png
1. content/practices/color-settings/images/currentcolor-windows11-theme-desert.png
1. content/practices/color-settings/images/currentcolor-windows11-theme-night-sky.png
1. content/practices/color-settings/images/currentcolor-windows11-theme-none.png
1. content/practices/color-settings/images/forced-colors-windows11-theme-desert.png
1. content/practices/color-settings/images/forced-colors-windows11-theme-night-sky.png
1. content/practices/color-settings/images/forced-colors-windows11-theme-none.png
1. content/practices/color-settings/images/input-range-element-aquatic-theme.png
1. content/practices/color-settings/images/input-range-element-desert-theme.png
1. content/practices/color-settings/images/input-range-element-night-sky-theme.png
1. content/practices/color-settings/images/input-text-element-aquatic-theme.png
1. content/practices/color-settings/images/input-text-element-desert-theme.png
1. content/practices/color-settings/images/input-text-element-night-sky-theme.png
1. content/practices/color-settings/images/p-element-aquatic-theme.png
1. content/practices/color-settings/images/p-element-desert-theme.png
1. content/practices/color-settings/images/p-element-night-sky-theme.png
1. content/practices/color-settings/images/radio-element-aquatic-theme.png
1. content/practices/color-settings/images/radio-element-desert-theme.png
1. content/practices/color-settings/images/radio-element-night-sky-theme.png
1. content/practices/color-settings/images/source/aquaic-screen-shot.png
1. content/practices/color-settings/images/source/desert-screen-shot.png
1. content/practices/color-settings/images/source/dusk-screen-shot.png
1. content/practices/color-settings/images/source/night-sky-screen-shot.png
1. content/practices/color-settings/images/switch-button-aquatic.png
1. content/practices/color-settings/images/switch-button-default.png
1. content/practices/color-settings/images/switch-button-desert.png
1. content/practices/color-settings/images/switch-button-dusk.png
1. content/practices/color-settings/images/switch-button-night-sky.png
1. content/practices/color-settings/images/switch-div-dusk.png
1. content/practices/color-settings/images/windows11-theme-none.png

mcking65 commented:
@jongund @daniel-montalvo 

The link checker is failing on this Medium link because it returns a 403:
https://medium.com/@tundehercules/designing-effective-dark-mode-interfaces-17f38ecea2e9

Medium is prompting for a sign in. You can read the article if you dismiss the sign in prompt. Nonetheless, it makes sense the link checker would fail.

The sign in is annoying. Then, there is another pop up later prompting you to subscribe.

If we want to reference Medium articles, it seems like we might have to add an exclusions input file for the link checker and add this link to it. Alternatively, we don't reference Medium. I would lean toward not referencing medium unless we believe this is a uniquely compelling reference that is worth significant additional maintenance effort.

howard-e commented:
> If we want to reference Medium articles, it seems like we might have to add an exclusions input file for the link checker and add this link to it. Alternatively, we don't reference Medium. I would lean toward not referencing medium ...

@mcking65 @daniel-montalvo this is pretty unintuitive but that would be possible right now by adding something like the following to [`.link-checker.js#excludedLinks` export](https://github.com/w3c/aria-practices/blob/main/.link-checker.js#L33):

```javascript
  excludedLinks: {
    'content/practices/color-settings/color-settings-practice.html': [
      'a[href^="https://medium.com/"]',
    ],
    // ...
```

> ... unless we believe this is a uniquely compelling reference that is worth significant additional maintenance effort

Something better is there is a `ignoreHashesOnExternalPagesMatchingRegex` export that's only ignoring hashes after a page has been loaded. Reworking that or having some `ignoreExternalPagesMatchingRegex` seems better suited instead of the above. That way, only minimal support could be added for that to [`link-checker.js`](https://github.com/w3c/aria-practices/blob/main/scripts/link-checker.js)

Either way, this config structure doesn't feel too intuitive and could maybe do with some future re-work.

@jongund [`color-settings-practice.html#829`](https://github.com/w3c/aria-practices/blob/8a828e84de224827a316d03ae949721f786a62ae/content/practices/color-settings/color-settings-practice.html#L829) looks to be correctly reporting that #do-not-use-system-colors can't be found as well. I'm guessing `System Colors Unsupported` is the target?

daniel-montalvo commented:
> If we want to reference Medium articles, it seems like we might have to add an exclusions input file for the link checker and add this link to it. Alternatively, we don't reference Medium. I would lean toward not referencing medium unless we believe this is a uniquely compelling reference that is worth significant additional maintenance effort.

I would +1 not referencing medium mainly because of these obscure patterns you mention, @mcking65.

mcking65 committed:
Simplify table structures

mcking65 committed:
Remove aria-labelledby pointing to body on system-colors table

mcking65 committed:
minor editorial changes for consistency

mcking65 commented:
@jongund 

Why is the list of unsupported system colors added via JavaScript?

mcking65 commented:
@jongund 

I am working on clarifying text in the section about testing contrast themes. I read the referenced chrome developer documentation about emulating CSS media features.

If you emulate forced-colors active in Chrome, which theme is rendered? Does that emulation work on platforms other than Windows? If so, what theme does it use?

mcking65 committed:
Editorial clarification of testing contrast themes

mcking65 committed:
Partial revision of instructions for supporting contrast schemes

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,936 @@
+            Choose color palettes that are easy to read in both light and dark modes.
+          </li>
+          <li>
+            The CSS <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark"><code>light-dark</code> function</a> or a CSS <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme">media query <code>prefers-color-scheme</code></a>,
```
@jongund
This line is a sentence fragment. Could you please complete this thought. Once the author has designed the color scheme, I assume step 2 is supporting it in code and that is what this is about.

jongund committed:
made tables static content and removed script reference for generating table information and color descriptions

jongund committed:
Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund committed:
fixed some linting errors

jongund committed:
updated documentation on computed  color column

jongund commented:
@mcking65 
Updated the pull request to use only static content in the tables.  
The computed color in the system color table is now static content based on  macOS and Chromium browser rendering.

Should I add another column with Firefox and Windows rendering?

mcking65 committed:
Delete link to medium and move resources into the supporting steps for contrast schemes

mcking65 committed:
Merge branch 'high-contrast-practice' of https://github.com/w3c/aria-practices into high-contrast-practice

jongund committed:
removed unsed images

jongund committed:
ø

Merge branch 'high-contrast-practice' of github.com:w3c/aria-practices into high-contrast-practice

jongund commented:
@mcking65 
Removed unused images

jongund committed:
fixed broken link

jongund committed:
fixed html liniting error

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1206 @@
+            <tr>
+              <th>System Color</th>
+              <th>Contrast Theme Color</th>
+              <th>Computed Color (Chromium)</th>
```
@jongund

Could we make two changes to this table?

1. Add a column for Firefox custom contrast that shows the name of the Firefox setting that corresponds to the system color.

2. Add rows for the other system colors so we have all colors in one table and can show the ones that are mapped in only one of the 3 (windows, chrome, or firefox) and which colors, if any, are not mapped to any of the 3.

mcking65 commented:
@jongund 

As I read more about contrast themes, given both the Firefox and Chrome support for forced colors, it no longer seems accurate to call this a Windows operating system feature. The Firefox support, especially with its ability to mirror Linux operating system settings and its customizability on macOS, is really bringing this feature to the web platform. Our content is not doing that work justice.

So, I am going to revise how we describe this feature. One part that I'd really like your help with is the table of system colors and how they are mapped by Windows, Firefox, and Chrome. I added a comment on the table itself. Can you do that? Is what I am suggesting clear and sensible?

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1198 @@
+            The following table shows the twenty-three system color keywords and indicates which are supported in the Windows, Firefox, and Chromium implementations.
+</p>
+<table class="data">
+          <caption>System Colors Supported by Various Contrast Theme Implementations</caption>
```
@jongund
This table is now moved and has its full context. Could you please complete this table as described by its new context?

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1198 @@
+          <li><a href="../../patterns/slider/examples/slider-rating.html">Rating Slider</a></li>
+        </ul>
+
+        <h3>Supporting Contrast Themes</h3>
```
@jongund

This section was missing. Could you please draft content for it that is similar in form to the section on supporting color schemes?

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1199 @@
+              <th>ActiveText</th>
+              <td>Hyperlink</td>
+              <td>
+                <div role="img" class="sample" aria-label="Active text is red" style="background-color: activetext;">
```
@jongund is it OK to change `activetext` to `activeText`, or do we need to add `activetext` to the cspell config?

mcking65 commented on content/practices/color-settings/color-settings-practice.html:
```diff
@@ -0,0 +1,1199 @@
+              <th>ButtonBorder</th>
+              <td>Button Text</td>
+              <td>
+                <div role="img" class="sample" aria-label="Button border is black" style="background-color: buttonborder;"></div>
+                <div class="color">#000000</div>
```
@jongundis it OK to change `buttonborder` to `buttonBorder`, or do we need to add `buttonborder` to the cspell config?
